sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -1937,35 +1953,68 @@ public void testNullEmptyStringEquality() throws Exception
   {
-    testQuery(
-        "SELECT COUNT(*)\n"
-        + "FROM druid.foo\n"
-        + "WHERE NULLIF(dim2, 'a') = ''",
-        ImmutableList.of(
-            Druids.newTimeseriesQueryBuilder()
-                  .dataSource(CalciteTests.DATASOURCE1)
-                  .intervals(querySegmentSpec(Filtration.eternity()))
-                  .granularity(Granularities.ALL)
-                  .filters(expressionFilter("case_searched((\"dim2\" == 'a'),"
-                                            + (NullHandling.replaceWithDefault() ? "1" : "0")
-                                            + ",(\"dim2\" == ''))"))
-                  .aggregators(aggregators(new CountAggregatorFactory("a0")))
-                  .context(TIMESERIES_CONTEXT_DEFAULT)
-                  .build()
-        ),
-        ImmutableList.of(
-            NullHandling.replaceWithDefault() ?
-            // Matches everything but "abc"
-            new Object[]{5L} :
-            // match only empty string
-            new Object[]{1L}
-        )
-    );
+    if (NullHandling.replaceWithDefault()) {
+      testQuery(
+          "SELECT COUNT(*)\n"
+          + "FROM druid.foo\n"
+          + "WHERE NULLIF(dim2, 'a') = ''",
+          ImmutableList.of(
+              Druids.newTimeseriesQueryBuilder()
+                    .dataSource(CalciteTests.DATASOURCE1)
+                    .intervals(querySegmentSpec(Filtration.eternity()))
+                    .granularity(Granularities.ALL)
+                    .filters(in("dim2", ImmutableList.of("", "a"), null))
+                    .aggregators(aggregators(new CountAggregatorFactory("a0")))
+                    .context(TIMESERIES_CONTEXT_DEFAULT)
+                    .build()
+          ),
+          ImmutableList.of(
+              // Matches everything but "abc"
+              new Object[]{5L}
+          )
+      );
+    } else {
+      testQuery(
+          "SELECT COUNT(*)\n"
+          + "FROM druid.foo\n"
+          + "WHERE NULLIF(dim2, 'a') = ''",
+          ImmutableList.of(
+              Druids.newTimeseriesQueryBuilder()
+                    .dataSource(CalciteTests.DATASOURCE1)
+                    .intervals(querySegmentSpec(Filtration.eternity()))
+                    .granularity(Granularities.ALL)
+                    .filters(selector("dim2", "", null))
+                    .aggregators(aggregators(new CountAggregatorFactory("a0")))
+                    .context(TIMESERIES_CONTEXT_DEFAULT)
+                    .build()
+          ),
+          ImmutableList.of(
+              // match only empty string
+              new Object[]{1L}
+          )
+      );
+    }
   }
 
   @Test
   public void testNullStringEquality() throws Exception
   {
+    testQuery(
+        "SELECT COUNT(*)\n"
+        + "FROM druid.foo\n"
+        + "WHERE NULLIF(dim2, 'a') = null",
+        ImmutableList.of(),
+        NullHandling.replaceWithDefault() ?
+        // Matches everything but "abc"
```

Comment:
Looks like the comment is wrong?

Reply:
Fixed this area, the ternary was also unnecessary